src/Spice86.Core/Emulator/CPU/ModRM.cs

```diff
@@ -18,48 +19,48 @@ public class ModRM {
         _state = state;
     }
 
-    public uint GetAddress(int defaultSegmentRegisterIndex, ushort offset) {
-        int? segmentIndex = _state.SegmentOverrideIndex;
+    public uint GetAddress(uint defaultSegmentRegisterIndex, ushort offset) {
+        uint? segmentIndex = _state.SegmentOverrideIndex;
         segmentIndex ??= defaultSegmentRegisterIndex;
 
-        ushort segment = _state.SegmentRegisters.GetRegister16((int)segmentIndex);
+        ushort segment = _state.SegmentRegisters.UInt16[(int)segmentIndex];
```

Comment:
Why is a `uint?` cast to an `int`?
I'd expect the segmentIndex to either already be an `int`, or the `segmentIndex.Value` being used.